### PR TITLE
Visit 2.13.0

### PR DESCRIPTION
**Pull request by goxberry**
* Add versions 2.12.3, 2.13.0

* Enable building with `hdf5+mpi` via variant

* Add `qwt` dependency

### CODE REVIEW

**Comment by goxberry** [comment]
This PR includes commits from PRs #6946 and #6943 for testing. Once both of these previous PRs are merged, I can rebase this PR to clean up the history. This PR also completes fixes for issues #6406 and #5152.

**Comment by goxberry** [comment]
Also, regarding the style, I'm not especially thrilled about adding the `+mpi` variant here; I didn't bother to try and think about what the appropriate dependencies of `~mpi` would be (probably `depends_on('netcdf~mpi', when='~mpi'`). The variant is sort of a hack to get around the lack of variant forwarding.

**Comment by adamjstewart** [comment]
I don't like the addition of the `+mpi` variant. You can already get what you want on the command line:
```
$ spack install visit ^hdf+mpi
```

**Comment by goxberry** [comment]
@adamjstewart I tried that, and spack rebuilt `hdf5` without `mpi` when executing `spack install visit ^hdf5+mpi`; I don't understand why that happens, but it does. Also, I need some way of propagating the `mpi` dependency of `hdf5+mpi` into the `visit` package so I can pass additional flags to CMake. If I'm building VisIt using `hdf5~mpi`, I don't need to add `-DVISIT_HDF5_MPI_DIR:PATH={0}.format(spec['hdf5'].prefix)` to `cmake_args`. I thought a conditional like `if self.spec['hdf5'].satisfies('+mpi')` would work, but it was not successful.

**Comment by adamjstewart** [comment]
> Also, I need some way of propagating the `mpi` dependency of `hdf5+mpi` into the `visit` package so I can pass additional flags to CMake. If I'm building VisIt using `hdf5~mpi`, I don't need to add `-DVISIT_HDF5_MPI_DIR:PATH={0}.format(spec['hdf5'].prefix)` to `cmake_args`. I thought a conditional like `if self.spec['hdf5'].satisfies('+mpi')` would work, but it was not successful.

You should be able to use:
```python
if 'hdf5+mpi' in self.spec:
```
or:
```python
if self.spec.satisfies('hdf5+mpi'):
```
@alalazo Wasn't the "`self.spec['hdf5'].satisfies('+mpi')` not working" thing a bug?

**Comment by adamjstewart** [comment]
> I tried that, and spack rebuilt `hdf5` without `mpi` when executing `spack install visit ^hdf5+mpi`; I don't understand why that happens, but it does.

In its current state, `visit` explicitly depends on `hdf5~mpi`. If I remove that constraint, it works as expected.

**Comment by alalazo** [comment]
@adamjstewart @goxberry I think you refer to #4135 . In that case the current solution is to write:
```python
if self.spec.satisfies('^hdf5+mpi', strict=True):
    pass
```
instead of
```python
if self.spec.satisfies('^hdf5+mpi'):
    pass
```
to search *just the `self.spec` node*. Saying:
```python
if '^hdf5+mpi' in self.spec:
    pass
```
works, but will search the entire DAG for a match.

File: var/spack/repos/builtin/packages/visit/package.py [hunk 1]
```diff
@@ -31,31 +31,43 @@ class Visit(CMakePackage):
...
 
+    variant('mpi', default=False, description='Enable HDF5 MPI support')
+
     depends_on('cmake@3.0:', type='build')
     depends_on('vtk@6.1.0~opengl2')
     depends_on('qt@4.8.6')
+    depends_on('qwt')
     depends_on('python')
     depends_on('silo+shared')
-    depends_on('hdf5~mpi')
```
[the remark this answers is not in the document]
Comment (goxberry) [reply var/spack/repos/builtin/packages/visit/package.py]: SGTM. Will rebase and clean up the history after the other two VisIt-related PRs are merged.

**Comment by goxberry** [comment]
@davydden @adamjstewart Rebased and should be ready to merge.

**Pull request merged**